plugin/storage/grpc/factory.go

```diff
@@ -66,10 +66,12 @@ func NewFactoryWithConfig(
 	metricsFactory metrics.Factory,
 	logger *zap.Logger,
 	host component.Host,
+	meterProvider metric.MeterProvider,
```

Comment:
Use the opportunity to simplify the signature:
```suggestion
	host component.Host,
	telset telemetry.Settings,
```